server.go

```diff
@@ -136,6 +136,9 @@ func (s *ssServer) loadConfig(filename string) error {
 		if !ok {
 			return fmt.Errorf("Only AEAD ciphers are supported. Found %v", keyConfig.Cipher)
 		}
+		if err := shadowsocks.CheckCipher(aead); err != nil {
```

Comment:
`shadowsocks.CheckCipher(aead)` doesn't read well. We are not checking if they are a valid Shadowsocks cipher, which it seems to imply. And we already tested if it's an Outline-supported cipher above.

I think what we want is to check if the cipher can be part of the cipherList. So maybe we check on `cipher.PushBack` instead?

Reply:
`cipherList.PushBack()` is a method on `list.List`, not `CipherList`, so we can't easily act there.  I've moved the check to `CipherList.Update()`.